forc-plugins/forc-client/src/op/call/parser.rs

```diff
@@ -0,0 +1,1557 @@
+        let enum_variants = EnumVariants::new(vec![
+            ("Active".to_string(), ParamType::Bool),
+            ("Pending".to_string(), ParamType::U64),
+        ])
+        .unwrap();
+        let token = Token::Enum((1u64, Token::U64(42), enum_variants).into());
+        let output = token_to_string(&token).unwrap();
+        assert_eq!(output, "(Pending:42)");
+
+        // enum (complex example) - variants with a struct that contains an enum and a vec that contains another enum with 2 variants
+        let inner_enum_variants = EnumVariants::new(vec![
+            ("Active".to_string(), ParamType::Bool),
+            ("Pending".to_string(), ParamType::U64),
+        ])
+        .unwrap();
+        let enum_variants = EnumVariants::new(vec![
+            (
+                "Input".to_string(),
+                ParamType::Struct {
+                    generics: vec![],
+                    name: "".to_string(),
+                    fields: vec![
+                        (
+                            "".to_string(),
+                            ParamType::Enum {
+                                name: "".to_string(),
+                                enum_variants: inner_enum_variants.clone(),
+                                generics: vec![],
+                            },
+                        ),
+                        (
+                            "".to_string(),
+                            ParamType::Vector(Box::new(ParamType::Enum {
+                                name: "".to_string(),
+                                enum_variants: inner_enum_variants.clone(),
+                                generics: vec![],
+                            })),
+                        ),
+                    ],
+                },
+            ),
+            ("Active".to_string(), ParamType::Bool),
+            ("Pending".to_string(), ParamType::U64),
+        ])
+        .unwrap();
+
+        // test active variant
+        let token = Token::Enum((1u64, Token::Bool(true), enum_variants.clone()).into());
+        let output = token_to_string(&token).unwrap();
+        assert_eq!(output, "(Active:true)");
+
+        // test Input variant
+        let token = Token::Enum(
+            (
+                0u64,
+                Token::Struct(vec![
+                    Token::Enum((0u64, Token::Bool(true), inner_enum_variants.clone()).into()),
+                    Token::Vector(vec![Token::Enum(
+                        (1u64, Token::U64(42), inner_enum_variants.clone()).into(),
+                    )]),
+                ]),
+                enum_variants,
+            )
+                .into(),
+        );
+        let output = token_to_string(&token).unwrap();
+        assert_eq!(output, "(Input:{(Active:true), [(Pending:42)]})");
+
+        // enum (complex example 2) - 2 variants, one with a struct that contains another struct with 2 fields, another with a struct
+        // enum GenericEnum<GenericStruct<u32>> {
```

Comment:
do we need this commented section still?

Reply:
i think it's useful to keep it there for debugging purposes to know what we are dealing with; however can remove if we want

Reply:
No worries, it's fine. If it's useful then leave it. Was just checking.